### PR TITLE
fix: spawn child process with process.env in macOS arm64

### PR DESCRIPTION
**Pull request by willmsC**
Add process.env to a child process identifying a custom browser on Apple Silicon.

### Additional details

In our organization we use a custom Chrome browser which requires a number of basic environment variables, e.g. HOME, to load necessary configs. 
When we try to run Cypress with this browser on Apple Silicon, it is not recognized. See log below: 

```
> npx cypress run --browser="/Users/cwillms/Projects/custom_chrome/bin/macos/custom_chrome"
We could not identify a known browser at the path you provided: /Users/cwillms/Projects/custom_chrome/bin/macos/custom_chrome

Read more about how to troubleshoot launching browsers: https://on.cypress.io/troubleshooting-launching-browsers

The output from the command we ran was:

Unable to find browser with path /Users/cwillms/Projects/custom_chrome/bin/macos/custom_chrome
We could not identify a known browser at the path you provided: /Users/cwillms/Projects/custom_chrome/bin/macos/custom_chrome

Read more about how to troubleshoot launching browsers: https://on.cypress.io/troubleshooting-launching-browsers

The output from the command we ran was:

Unable to find browser with path /Users/cwillms/Projects/custom_chrome/bin/macos/custom_chrome
```

However, the browser is returning a correct version information when called:
```
> /Users/cwillms/Projects/custom_chrome/bin/macos/custom_chrome --version
Google Chrome 106.0.0
```

A detailed investigation showed that in the childprocess started in utils.ts (lines 19 to 29) only the `ARCHPREFERENCE` variable set in line 25 is available, but not the ones we need. For this reason, our custom browser could not be called in the childprocess and therefore could not return its versions information. I suggest to include also the process.env in this childprocess. This solved the problem for us. 

When calling the childprocess for non Apple Silicon in line 33 this is already done implicitly, because no dedicated variables are set here. 

### PR Tasks

- [x] Have tests been added/updated?
- [x] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [n/a] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [n/a] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Comment by mschile** [comment]
Hi @willmsC 👋, thanks for this contribution! I added a few tests to verify the environment variables are correctly passed.

**Review by mschile — APPROVED** [review]

**Review by flotwig — APPROVED** [review]
@willmsC Thank you for the fix. This only start happening for you in 12.3.0, correct? #25014 seems like it would cause exactly this.

File: cli/CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ _Released 02/14/2023 (PENDING)_
 
 - Fixed an issue with the Cloud project selection modal not showing the correct prompts. Fixes [#25520](https://github.com/cypress-io/cypress/issues/25520).
 - Fixed an issue in middleware where error-handling code could itself generate an error and fail to report the original issue. Fixes [#22825](https://github.com/cypress-io/cypress/issues/22825).
+- Fixed an issue where custom browsers may not be found in macOS arm64 architectures that rely on process environment variables. Fixed in [#25753](https://github.com/cypress-io/cypress/pull/25753).
```
Comment (emilyrohrbough) [inline cli/CHANGELOG.md]:
```suggestion
- Fixed an issue where custom browsers that relied on process environment variables were not found on macOS arm64 architectures. Fixed in [#25753](https://github.com/cypress-io/cypress/pull/25753).
```

**Review by emilyrohrbough — APPROVED** [review]

**Comment by willmsC** [comment]
@flotwig Yes, exactly. The changes in https://github.com/cypress-io/cypress/pull/25014 do indeed look like they are causing the issue.

**Review by ryanthemanuel — APPROVED** [review]

**Pull request merged**